### PR TITLE
feat: improve Operator performance by using caching to reduce api call  and network impact

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

split origin PR https://github.com/opendatahub-io/opendatahub-operator/pull/1048 into 2 parts.
this PR is part II, focusing on add support in cache option.
(work based on https://github.com/opendatahub-io/opendatahub-operator/pull/1189)
- secret: default application namespace + other default ones + istio cert
- configmap: all
- namespace: all
- ingressctrler: "default" one
- deployment: default application namespaces + default namespaces

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by ykaliuta** [comment]
But webhook changes do not anyhow connected (and reflected) in the header, "feat: improve Operator performance by using caching to reduce api call and network impact", or I'm missing something?

**Comment by zdtsw** [comment]
> But webhook changes do not anyhow connected (and reflected) in the header, "feat: improve Operator performance by using caching to reduce api call and network impact", or I'm missing something?

no, i need to get the webhook out as well. 
it is the "rest" from the original PR, so i did split them to 3 pieces but onlyl 2, thats why webhook was kept here

File: config/default/manager_webhook_patch.yaml [hunk 1]
```diff
@@ -20,4 +20,4 @@ spec:
       - name: cert
         secret:
           defaultMode: 420
-          secretName: opendatahub-operator-controller-webhook-cert
+          secretName: opendatahub-operator-webhook-service
```
Comment (ykaliuta) [inline config/default/manager_webhook_patch.yaml:23]: So why is this change?  I mean
1) why to rename?
2) I see service.beta.openshift.io/serving-cert-secret-name: opendatahub-operator-webhook-cert in Service and would expect the same here to mount as a volume but looks like there is some other logic behind it now. In the bundle it is different. I see in the message "- secret: default application namespace + other default ones + istio cert" Could you share some link?

Excuse my ignorance :)

Comment (zdtsw) [reply config/default/manager_webhook_patch.yaml]: i am trying to remember the reason why it was changed.
will need to dig into the old commit and find what exactly happen, but was something related to the bundle csv and the way it creates service for webhook which does not match the one if we test locally for e2e.

File: main.go [hunk 2]
```diff
@@ -285,3 +318,37 @@ func main() { //nolint:funlen
 		os.Exit(1)
 	}
 }
+
+func setSecretCache(platform cluster.Platform) map[string]cache.Config {
```
Comment (ykaliuta) [inline main.go:322]: I have the same concern about naming. For my opinion it does not "set" anything, but creates the config. I saw in other projects that using "new" for such cases is idiomatic for golang as well, but not "set".

Comment (ykaliuta) [reply main.go:322]: createSecretCacheConfig() sounds clear for me

File: main.go [hunk 3]
```diff
@@ -285,3 +318,37 @@ func main() { //nolint:funlen
...
+		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
+	case cluster.SelfManagedRhods:
+		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
+	default:
+		namespaceConfigs["opendatahub"] = cache.Config{}
+	}
+	return namespaceConfigs
+}
+
+func createDeploymentCache(platform cluster.Platform) map[string]cache.Config {
```
Comment (ykaliuta) [inline main.go:339]: createDeploymentCacheConfig as above? Since it doesn't create the Cache, but just its config.

Comment (zdtsw) [reply main.go:339]: thought i updated it to setDeploymentCache last week? or i did not push it :thinking:

Comment (zdtsw) [reply main.go]: seems, it does only live in my local.............

**Review by ykaliuta — APPROVED** [review]

**Pull request merged**